Make it easy to link static websites without public folder

stefansundin opened the pull request:
This is an alternative to https://github.com/powder-rb/powder/pull/115. I think it's a cleaner solution.

stefansundin committed:
Make it easy to link static websites.

stefansundin committed:
Fix unlink for static website.

philnash commented:
To be clear here. This creates the public directory under `~/.pow/dir_name` and that then works for static sites in the folder?

What needed updating in the `unlink` command? Why might there be an error saying that the symlink couldn't be removed?

stefansundin commented:
Correct. It creates `~/.pow/dir_name`, and then creates a symlink from `~/.pow/dir_name/public` to the current directory. Basically emulating `public/` without requiring you to create a public directory in your project.

I need to update unlink because `rm_f` is unable to delete `~/.pow/dir_name` if it is a directory. The reason I added the check is to not say "successfully" if we just assume the previous call worked. Before I changed `rm_f` to `rm_rf`, it said that it was successful but it wasn't.
